In [3]:
import pandas as pd

crime = pd.read_csv("datasets/Crime_Data_from_2020_to_Present.csv")

In [4]:
from pandasql import sqldf

In [5]:
print(sqldf('''
            SELECT "TIME OCC", "Crm Cd Desc"
            FROM crime
            '''))

        TIME OCC                                        Crm Cd Desc
0           2130                                   VEHICLE - STOLEN
1           1800                              BURGLARY FROM VEHICLE
2           1700                                      BIKE - STOLEN
3           2037           SHOPLIFTING-GRAND THEFT ($950.01 & OVER)
4           1200                                  THEFT OF IDENTITY
...          ...                                                ...
984040      1400                                   VEHICLE - STOLEN
984041       100           VANDALISM - MISDEAMEANOR ($399 OR UNDER)
984042      2330  THEFT-GRAND ($950.01 & OVER)EXCPT,GUNS,FOWL,LI...
984043      1500     ASSAULT WITH DEADLY WEAPON, AGGRAVATED ASSAULT
984044      2300                                   VEHICLE - STOLEN

[984045 rows x 2 columns]


In [6]:
print(sqldf("""
            SELECT DISTINCT "Crm Cd Desc"
            FROM crime
            """))

                                  Crm Cd Desc
0                            VEHICLE - STOLEN
1                       BURGLARY FROM VEHICLE
2                               BIKE - STOLEN
3    SHOPLIFTING-GRAND THEFT ($950.01 & OVER)
4                           THEFT OF IDENTITY
..                                        ...
135                           INCITING A RIOT
136        DISHONEST EMPLOYEE ATTEMPTED THEFT
137                       FAILURE TO DISPERSE
138                            TRAIN WRECKING
139                      DRUNK ROLL - ATTEMPT

[140 rows x 1 columns]


All the unique crimes from 2020-now.

In [7]:
print(sqldf( """
            SELECT DISTINCT "Crm Cd Desc"
            FROM crime
            WHERE "Crm Cd Desc" LIKE "%STOLEN"
            """))

                Crm Cd Desc
0          VEHICLE - STOLEN
1             BIKE - STOLEN
2  VEHICLE - ATTEMPT STOLEN
3             BOAT - STOLEN
4   BIKE - ATTEMPTED STOLEN


These were items that were either stolen or attempted at stealing from 2020 - now.

In [8]:
print(sqldf( """
            SELECT COUNT("Crm Cd Desc")
            FROM crime
            WHERE "Crm Cd Desc" LIKE "%STOLEN"
            """))

   COUNT("Crm Cd Desc")
0                123043


In LA between 2020 and now, there have been 123,043 things stolen, whethere that range from a bike to a boat and if it was an attempt or successful. 

In [11]:
print(sqldf( """
            SELECT DISTINCT "Crm Cd Desc", "Time OCC"
            FROM crime
            WHERE "Crm Cd Desc" LIKE "%STOLEN"
            AND "Time OCC" BETWEEN 0900 AND 1000
            """))

                 Crm Cd Desc  TIME OCC
0           VEHICLE - STOLEN      1000
1           VEHICLE - STOLEN       950
2   VEHICLE - ATTEMPT STOLEN       900
3           VEHICLE - STOLEN       900
4              BIKE - STOLEN       935
..                       ...       ...
86          VEHICLE - STOLEN       906
87             BIKE - STOLEN       939
88          VEHICLE - STOLEN       919
89  VEHICLE - ATTEMPT STOLEN       938
90          VEHICLE - STOLEN       901

[91 rows x 2 columns]


In [12]:
print(sqldf( """
            SELECT DISTINCT "Crm Cd Desc", "Time OCC"
            FROM crime
            WHERE "Crm Cd Desc" LIKE "%THEFT"
            GROUP BY "Crm Cd Desc" HAVING "Time OCC" > 900
            """))

                          Crm Cd Desc  TIME OCC
0                  BUNCO, GRAND THEFT      1800
1    DISHONEST EMPLOYEE - GRAND THEFT      1320
2    DISHONEST EMPLOYEE - PETTY THEFT      1400
3  DISHONEST EMPLOYEE ATTEMPTED THEFT      1900


In [15]:
#im unsure of why it wont work
print(sqldf( """
            SELECT "Crm Cd Desc", "Time OCC"
            FROM crime
            WHERE "Time OCC" > (
            SELECT "Time OCC" FROM crime
                  WHERE "Time OCC" BETWEEN 1300 AND 1400
            )
            """))


                                              Crm Cd Desc  TIME OCC
0                                        VEHICLE - STOLEN      2130
1                                   BURGLARY FROM VEHICLE      1800
2                                           BIKE - STOLEN      1700
3                SHOPLIFTING-GRAND THEFT ($950.01 & OVER)      2037
4                                       THEFT OF IDENTITY      2300
...                                                   ...       ...
499608  THEFT-GRAND ($950.01 & OVER)EXCPT,GUNS,FOWL,LI...      2300
499609  THEFT-GRAND ($950.01 & OVER)EXCPT,GUNS,FOWL,LI...      1546
499610  THEFT-GRAND ($950.01 & OVER)EXCPT,GUNS,FOWL,LI...      2330
499611     ASSAULT WITH DEADLY WEAPON, AGGRAVATED ASSAULT      1500
499612                                   VEHICLE - STOLEN      2300

[499613 rows x 2 columns]
